src/extra/Markdown/index.tsx

```diff
@@ -138,11 +139,110 @@ type components = {
 	[element: string]: (props: any) => JSX.Element | React.ReactElement | Element;
 };
 
+interface Decorator {
+	match: RegExp;
+	component: string;
+}
+
+interface DecoratorPluginOptions {
+	decorators: Decorator[];
+}
+
+const decoratorPlugin = (options: DecoratorPluginOptions) => {
+	options = options || { decorators: [] };
+
+	if (!options.decorators || !options.decorators.length) {
+		return;
+	}
+
+	const targets: Array<{ decorator: Decorator; node: any }> = [];
+
+	const visitor = (node: any) => {
+		const matchedDecorator = options.decorators.find((decorator) => {
+			return (
+				node.children &&
+				node.children[0] &&
+				node.children[0].type === 'text' &&
+				decorator.match.test(node.children[0].value)
+			);
+		});
+
+		if (matchedDecorator) {
+			targets.push({
+				decorator: matchedDecorator,
+				node,
+			});
+		}
+	};
+
+	const modify = () => {
+		targets.forEach((target) => {
+			const node = target.node;
+			const match = target.decorator.match;
+
+			const text = node.children[0].value;
+			const children = [];
+
+			match.lastIndex = 0;
+			do {
+				const lastIndex = match.lastIndex;
+				const result = match.exec(text);
+
+				if (result) {
+					children.push(
+						{
+							type: 'element',
+							tagName: 'span',
+							children: [
+								{
+									type: 'text',
+									value: text.substring(lastIndex, result.index),
+								},
+							],
+						},
+						{
+							type: 'element',
+							tagName: target.decorator.component,
+							children: [
+								{
+									type: 'text',
+									value: result[0],
+								},
+							],
+						},
+					);
+				} else {
+					children.push({
+						type: 'element',
+						tagName: 'span',
+						children: [
+							{
+								type: 'text',
+								value: text.substring(lastIndex),
+							},
+						],
+					});
+
+					break;
+				}
+			} while (match.global);
+
+			node.children = children;
+		});
+	};
+
+	return (tree: any) => {
+		visit(tree, null, visitor);
```

Comment:
We need to measure the performance impact of traversing the entire tree, and see how it impacts JF in particular.